### PR TITLE
translated "API" page into Korean

### PR DESCRIPTION
**Pull request by jiyeonseo**
API page into Korean 🇰🇷 

related issue : https://github.com/react-hook-form/react-hook-form-website/issues/42

### CODE REVIEW

**Comment by jiyeonseo** [comment]
@iamchanii @adhrinae Could you take a look? :)

**Comment by bluebill1049** [comment]
awesome!! thanks @jiyeonseo we can go live with Korean documentation once this gets merged <3 super happy. thanks a lot <3

File: src/data/api.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default {
       en:
         "React Hook Form focuses on providing the best DX by simplifying the API.",
       kr:
-        "React Hook Form focuses on providing the best DX by simplifying the API.",
+        "React Hook Form는 API 단순화를 통해 최고의 DX제공에 집중합니다.",
```
Comment (adhrinae) [inline src/data/api.tsx]: "최고의 DX를 제공합니다"

File: src/data/api.tsx [hunk 2]
```diff
@@ -156,132 +156,95 @@ export default {
       title: "useForm",
       intro: (
         <>
-          By invoking <code>useForm</code>, you will receive the following
-          methods{" "}
+          <code>useForm</code> 호출을 통해, 다음과 같은 메소드들을 제공합니다.{" "}
```
Comment (adhrinae) [inline src/data/api.tsx]: "useForm 을 호출하려 다음의 메서드를 사용할 수 있습니다"

File: src/data/api.tsx [hunk 3]
```diff
@@ -156,132 +156,95 @@ export default {
...
+          <code>useForm</code> 에서는 <b>옵셔널</b>한 인수도 있습니다. 다음 예제는 모든 옵션의 기본값을 보여줍니다.
         </p>
       ),
       validationSchemaOption: (
         <>
-          The <code>options</code> argument is an object hash containing any
-          schema options you may want to override (or specify for the first
-          time). You can read more in the{" "}
-          <a
+          <code>옵션</code>인수는 재정의(또는 처음 지정)할 수 있는 스키마 옵션이 포함된 개체 해시입니다. 더 자세한 사항은 <a
```
Comment (adhrinae) [inline src/data/api.tsx]: "스키마 옵션이 포함된 객체 해시입니다"

File: src/data/api.tsx [hunk 4]
```diff
@@ -156,132 +156,95 @@ export default {
...
+          <code>blur</code> 이벤트에서 유효성 검사가 트리거됩니다.
         </>
       ),
       validateOnChange: (
         <>
-          Validation will trigger on the <code>change</code> event with each
-          input, and lead to multiple re-renders. Not recommended: Consider this
-          as a bad performance practice.
+          유효성 검사는 각 입력창의 <code>change</code> 이벤트로 시작되어, 여러번 다시 렌더링합니다. 
+          이 방법은 성능을 저해함으로 추천하지 않습니다.
```
Comment (adhrinae) [inline src/data/api.tsx]: "이 방법은 랜더링 성능을 떨어뜨리므로 추천하지 않습니다."

File: src/data/api.tsx [hunk 5]
```diff
@@ -156,132 +156,95 @@ export default {
...
-          are required to validate.
+          제공되는 필드 배열은 앞으로 검증될 필드만 포함합니다. 이 옵션은 유효성 검사가 필요한 필드에 대해서 토글(toggle)을 원할 경우에 유용합니다. 
         </p>
       ),
       submitFocusError: (
         <>
           <p>
-            By default when the user submits a form and that contains an error,
-            the first field with an error will be focused.
+            기본적으로 사용자가 폼을 제출하고 에런가 있는 경우, 에러가 있는 첫번째 필드에 포커스 됩니다.
```
Comment (adhrinae) [inline src/data/api.tsx]: "제출하고 에러가 있는 경우"

File: src/data/api.tsx [hunk 6]
```diff
@@ -1589,15 +1515,12 @@ export default {
...
           <p>
-            <Note>Note:</Note> By invoking this method, <code>formState</code>{" "}
-            will push the input's <code>name</code> into <code>touched</code>.
+            <Note>참고:</Note> 이 함수를 호출함으로써, <code>formState</code>는 이 입력의 <code>name</code>을 <code>touched</code>로 설정하게 됩니다.
           </p>
           <p>
-            You can also set <code>shouldValidate</code> to <code>true</code>{" "}
-            and it will trigger field validation. eg:{" "}
-            <code>setValue('name', 'value', true)</code>
+            <code>shouldValidate</code>를 <code>true</code>로 설정하여, 필드 유효성 검사를 트리거할 수 도 있습니다. 예를 들어, <code>setValue('name', 'value', true)</code>
```
Comment (adhrinae) [inline src/data/api.tsx]: 예를 들어 -> `예: ` 정도면 되겠습니다.

**Comment by bluebill1049** [comment]
merge when you are ready @jiyeonseo

**Comment by adhrinae** [comment]
@bluebill1049 Need to change more `Link` components before the deployment.

After I changed the language and click any links in the Korean pages, it redirects to the English page, preserving dropdown selection.

**Comment by bluebill1049** [comment]
good catch @adhrinae, @jiyeonseo  I will merge this PR and fix it :)

**Comment by bluebill1049** [comment]
which link @adhrinae ?

**Comment by bluebill1049** [comment]
@adhrinae when you select from the drop, which link did you click which went to the English doc? https://react-hook-form.com/kr

**Comment by adhrinae** [comment]
@bluebill1049 I'll take care of them 🤟

**Comment by bluebill1049** [comment]
amazing! thank you @adhrinae

**Comment by bluebill1049** [comment]
there is a `translateLink` function FYI

**Pull request merged**